code/obj/machinery/traymachines.dm

```diff
@@ -0,0 +1,683 @@
+	var/id = 2 //this gets used when the tanning computer links to the bed
+	powerdraw_use = 1000 //power cost while tanning
+	mats = 30
+
+	icon_trayopen = "tanbed"
+	icon_unoccupied = "tanbed"
+	icon_occupied = "tanbed1"
+	tray_type = /obj/machine_tray/tanning
+
+	var/emagged = FALSE //heh heh
+	var/settime = 10 //How long? (s)
+	var/tanningcolor = rgb(205,88,34) //Change to tan people into hillarious colors!
+	var/tanningmodifier = 0.03 //How fast do you want to go to your tanningcolor?
+	var/obj/machinery/computer/tanning/linked = null
+
+	New()
+		..()
+		START_TRACKING
+
+	disposing()
+		src.linked?.linked = null
+		src.linked = null
+		. = ..()
+		STOP_TRACKING
+
+	emag_act(var/mob/user, var/obj/item/card/emag/E)
+		if (src.emagged)
+			return FALSE
+		if (user)
+			user.show_text("\The [src]'s safetey lock has been disabled.", "red")
+		src.emagged = TRUE
+		src.tanningmodifier = 0.2
+		return TRUE
+
+	proc/cremate(mob/user as mob)
+		if (!src || !istype(src))
+			return
+		if (status & NOPOWER)
+			return
+		if (src.locked)
+			return //don't let you cremate something twice or w/e
+		if (!src.contents || !length(src.contents))
+			src.visible_message("<span class='alert'>You hear the lights turn on for a second, then turn off.</span>")
+			return
+
+		src.visible_message("<span class='alert'>You hear a faint buzz as \the [src] activates.</span>")
+		playsound(src.loc, "sound/machines/shieldup.ogg", 30, 1)
+		src.locked = TRUE
+		power_usage = powerdraw_use
+		icon_state = "tanbed_active"
+
+		for (var/mob/M in contents)
+			if (isliving(M))
+				var/mob/living/L = M
+				for (var/i in 1 to src.settime)
+					sleep(1 SECOND)
+					if(ishuman(L))
+						var/mob/living/carbon/human/H = L
+						if (src.emagged)
+							H.TakeDamage("All", 0, 10, 0, DAMAGE_BURN)
+							if (src.settime % 2) //message limiter
+								boutput(H, "<span class='alert'>Your skin feels like it's on fire!</span>")
+						else if (!H.wear_suit)
+							H.TakeDamage("All", 0, 2, 0, DAMAGE_BURN)
+							if (src.settime % 2) //limiter
+								boutput(H, "<span class='alert'>Your skin feels hot!</span>")
+						if (!(H.glasses && istype(H.glasses, /obj/item/clothing/glasses/sunglasses))) //Always wear protection
+							H.take_eye_damage(1, 2)
+							H.change_eye_blurry(2)
+							H.changeStatus("stunned", 1 SECOND)
+							H.change_misstep_chance(5)
+							boutput(H, "<span class='alert'>Your eyes sting!</span>")
+						if (H.bioHolder.mobAppearance.s_tone)
+							var/currenttone = H.bioHolder.mobAppearance.s_tone
+							var/newtone = BlendRGB(currenttone, src.tanningcolor, src.tanningmodifier) //Make them tan slowly
+							H.bioHolder.mobAppearance.s_tone = newtone
+							H.set_face_icon_dirty()
+							H.set_body_icon_dirty()
+							if (H.limbs)
+								H.limbs.reset_stone()
+							H.update_colorful_parts()
+				if (emagged && isdead(M))
+					qdel(M)
+					make_cleanable( /obj/decal/cleanable/ash,src)
+
+		SPAWN_DBG(src.settime SECONDS)
```

Comment:
this should be set on the other side, not here

Reply:
I think the last push does what you mean?